src/main/java/org/mockito/internal/matchers/CapturingMatcher.java

```diff
@@ -26,16 +28,17 @@ public String toString() {
     }
 
     public T getLastValue() {
-        if (arguments.isEmpty()) {
-            throw noArgumentValueWasCaptured();
-        }
-
-        return (T) arguments.getLast();
+        synchronized (arguments) {
```

Comment:
If you think about it, it makes perfect sense. When you publish your collection, you want to make sure that everyone use the same lock when accessing your collection. That mean you have to pass the lock instance along with your collection reference. But since every object can be a synchronization primitive, you don't need to do that - you just synchronize on a collection itself to make operations serial.